sql/field.h

```diff
@@ -566,10 +566,19 @@ class Proto_field
   virtual bool send_text(Protocol *protocol)= 0;
 };
 
-class Field: public Proto_field
+/* auxiliary class to solve the issue with gcc-9 warning -Wdeprecated-copy */
+class nonassignable
+{
+protected:
+    nonassignable() {}
+    ~nonassignable() {}
+    nonassignable( const nonassignable& ) {}
+private:  /* emphasize the following member is private */
+    nonassignable& operator=( const nonassignable& );
+};
+
+class Field: public Proto_field, public nonassignable
```

Comment:
Please, reorder the classes being inherited. If you have `nonassignable` (an empty class) going first in the list the compiler will be able to use empty base class optimization and therefore the size of the `Field` class will not grow.
```suggestion
class Field: private nonassignable, public Proto_field
```